src/main/java/core/basesyntax/service/operation/ReturnOperation.java

```
@@ -0,0 +1,18 @@
+package core.basesyntax.service.operation;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.OperationHandler;
+import core.basesyntax.storage.Storage;
+
+public class ReturnOperation implements OperationHandler {
+    private final Storage storage;
+
+    public ReturnOperation(Storage storage) {
+        this.storage = storage;
+    }
+
+    @Override
+    public void handle(FruitTransaction transaction) {
```

The handler method does not validate its input. This violates the requirement to handle null parameters and invalid input data (e.g., "Null parameters"). Add an explicit null-check at the start of handle(...):
- if (transaction == null) throw new FruitShopException("Transaction is null for ReturnOperation");
Validating public method parameters and failing fast with informative errors is required.